ledger/participant-state/kvutils/tools/src/test/scala/com/daml/Replay.scala

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.test
+
+import java.io.DataInputStream
+import java.lang.System.err.println
+import java.nio.file.{Files, Path, Paths}
+import java.util.concurrent.TimeUnit
+
+import com.daml.ledger.participant.state.kvutils.Conversions._
+import com.daml.ledger.participant.state.kvutils.export.FileBasedLedgerDataExporter.SubmissionInfo
+import com.daml.ledger.participant.state.kvutils.export.Serialization
+import com.daml.ledger.participant.state.kvutils.{Envelope, DamlKvutils => Proto}
+import com.daml.ledger.participant.state.v1.ParticipantId
+import com.daml.lf.archive.{Decode, UniversalArchiveReader}
+import com.daml.lf.crypto
+import com.daml.lf.data._
+import com.daml.lf.engine.Engine
+import com.daml.lf.transaction.Node.GlobalKey
+import com.daml.lf.transaction.{Node, Transaction => Tx, TransactionCoder => TxCoder}
+import com.daml.lf.value.Value.ContractId
+import com.daml.lf.value.{Value, ValueCoder => ValCoder}
+import com.google.protobuf.ByteString
+import org.openjdk.jmh.annotations._
+
+import scala.collection.JavaConverters._
+
+final case class TxEntry(
+    tx: Tx.SubmittedTransaction,
+    participantId: ParticipantId,
+    ledgerTime: Time.Timestamp,
+    submissionTime: Time.Timestamp,
+    submissionSeed: crypto.Hash,
+)
+
+final case class BenchMarkSate(
```

Comment:
You're missing a "t", and I think the "m" should be lowercase.

```suggestion
final case class BenchmarkState(
```

Reply:
good catch.